README.md

```diff
@@ -27,143 +27,124 @@ The diagram below illustrates high-level design of each component as well as bas
 
 ```sh
-miden-node init
-```
-To generate the genesis block, run:
+# Install from a specific branch
+cargo install --locked --path bin/miden --git https://github.com/0xPolygonMiden/miden-node --branch <branch>
 
-```sh
-miden-node make-genesis
+# Install a specific tag
+cargo install --locked --path bin/miden --git https://github.com/0xPolygonMiden/miden-node --tag <tag>
+
+# Install a specific git revision
+cargo install --locked --path bin/miden --git https://github.com/0xPolygonMiden/miden-node --rev <git-sha>
 ```
 
-By default this will generate 1 file and 1 folder in the current directory:
+More information on the various options can be found [here](https://doc.rust-lang.org/cargo/commands/cargo-install.html#install-options).
 
-- `genesis.dat`: the genesis file.
-- `accounts` directory containing `.mac` files (one per account) for the accounts defined in the genesis inputs file. Each `.mac` file contains full serialization of an account, including code, storage, and authentication info.
+> [!TIP]
+> Miden account generation uses a proof-of-work puzzle to prevent DoS attacks. These puzzles can be quite expensive, especially for test purposes. You can lower the difficulty of the puzzle by appending `--features testing` to the `cargo install ..` invocation. For example:
+> ```sh
+> cargo install miden-node --locked --features testing
+> ```
 
-### Running the node
-
-Using the node configuration file created in the previous step, start the node:
+You can verify the installation by checking the node's version:
 
 ```sh
-mkdir -p /opt/miden
-miden-node start --config <path-to-config-file> <component-to-be-started>
+miden-node --version
 ```
 
-Note that the `store.genesis_filepath` field in the config file must point to the `genesis.dat` file that you generated in the previous step.
-
-### Running the node as separate components
-
-If you intend on running the node as different processes, you will need to install and run each component separately.
-Please, refer to each component's documentation:
-
-- [RPC](crates/rpc/README.md#usage)
-- [Store](crates/store/README.md#usage)
-- [Block Producer](crates/block-producer/README.md#usage)
-
-Each directory containing the executables also contains an example configuration file. Make sure that the configuration files are mutually consistent. That is, make sure that the URLs are valid and point to the right endpoint.
-
-### Running the node using Docker
-
-If you intend on running the node inside a Docker container, you will need to follow these steps:
-
-1. Build the docker image from source
-
-   ```sh
-   make docker-build-node
-   ```
-
-   This command will build the docker image for the Miden node and save it locally.
+## Usage
 
-2. Run the Docker container
+### Setup
```

Comment:
I don't remember for sure, but I think when installing a debian package some default location is used for storage and config files. Maybe it is worth mentioning it here somewhere?

Reply:
Ah. Yeah it automagically sets up a systemd service which has expected locations.

The package also copies over the `toml` configuration files which I don't quite like since one still has to run `make-genesis` anyway, making the initialization only half complete.

I think I'll remove this so that normal installation and configuration are the same for all ways?

Reply:
Hmm. Actually I'm unsure now. I see that the post-install script adds a `miden` user but I have no idea why -- I'm not that clued up on this sort of devops.

Could be that its required for file permissions in which case creating the files afterwards wouldn't be correct? i.e. `make-genesis` won't really work.

😕

Reply:
Ah. The systemd file expects the `user` to be `miden` which is probably the only reason it was added. I hope.